pyomo/core/expr/numeric_expr.py

```diff
@@ -1282,8 +1293,15 @@ def __init__(self, args=None):
             self.linear_vars = args[n+1:]
         else:
             self.constant = 0
+            if constant is not None:
+                self.constant = constant
             self.linear_coefs = []
+            if linear_coefs is not None:
+                self.linear_coefs = linear_coefs
             self.linear_vars = []
+            if linear_vars is not None:
+                self.linear_vars = linear_vars
```

Comment:
For efficiency (to avoid making and disposing of an empty list), I would make these all along the lines of:
```
if linear_vars is None:
    self.linear_vars = []
else:
    self.linear_vars = linear_vars
```
Also, are there any concerns around `linear_vars` and `linear_ceofs` arguments being shared with whomever constructed this object, or is the API contract that the caller is responsible for making sure the lists aren't shared?